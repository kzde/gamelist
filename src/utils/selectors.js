
export const getGameDetailsByName = (games, name) => games.find(game => game.name === name);
export const getGamesByPlatform = (games, platform) => games.filter(game => game.platform === platform);