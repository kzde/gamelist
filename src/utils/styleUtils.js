import { css } from 'styled-components';

export const colorPalettes = {
  black: '#333333',
  white: '#FFFFFF',
  red: '#FE6050',
  lightGrey: '#E5E5E5'
};

export const themeColors = {
  default: colorPalettes.black,
  selected: colorPalettes.red,
  link: colorPalettes.red,
  borderGrey: '#C4C4C4'
};

export const textSizes = {
  smallText: '14px',
  normalText: '18px',
  titleText: '24px'
};

export const margins = {
  small: '8px',
  medium: '16px',
  large: '32px'
};

export const headerHeight = '64px';
export const basicMargin = 16;
export const cardSize = {
  width: '328px',
  height: '492px',
  phoneWidth: '359px'
};

const sizes = {
  giant: 1170,
  desktop: 992,
  tablet: 768,
  phone: 576
};

export const media = Object.keys(sizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (max-width: ${sizes[label] / 16}em) {
      ${css(...args)}
    }
  `;
  return acc;
}, {});
