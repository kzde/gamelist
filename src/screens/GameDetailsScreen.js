import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Header from '../components/Header';
import { getGameDetailsByName } from '../utils/selectors';
import gamesObj from '../datas/games.json';
import Paper from '../components/Paper';
import DetailContent from '../components/DetailContent';
import NotFoundScreen from './NotFoundScreen';
import withI18n from '../contexts/withI18n';
import { media } from '../utils/styleUtils';

const { games } = gamesObj;

const Media = styled.div``;
const Img = styled.img`
  width: 100%;
  height: 368px;
  background: url(${props => props.desktopSrc});
  background-repeat: no-repeat;
  ${media.phone`
    background: url(${props => props.mobileSrc});
    background-repeat: no-repeat;
    background-position:center;
  `}
  background-position:center;
`;
const Shadow = styled.div`
  position: absolute;
  width: 100%;
  height: 368px;
  box-shadow: inset 0px -16px 16px rgba(0, 0, 0, 0.25);
  top: 64px;
  left: 0;
`;
const DetailSection = styled.section``;

const GameDetailsScreen = (props) => {
  const { gameName } = props.match.params;
  const gameDetails = getGameDetailsByName(games, gameName);
  return (
    <Fragment>
      {gameDetails ? (
        <Fragment>
          <Header text={props.i18n._('detail.header.title')} />
          <Media>
            <Img desktopSrc={gameDetails.hero} mobileSrc={gameDetails.cover} />
            <Shadow />
          </Media>
          <DetailSection>
            <Paper>
              <DetailContent
                title={gameDetails.name}
                subTitle={gameDetails.platform}
                content={gameDetails.description}
              />
            </Paper>
          </DetailSection>
        </Fragment>
      ) : (
        <NotFoundScreen />
      )}
    </Fragment>
  );
};

GameDetailsScreen.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      gameName: PropTypes.string.isRequired
    })
  }).isRequired,
  i18n: PropTypes.object.isRequired
};

export default withI18n(GameDetailsScreen);
