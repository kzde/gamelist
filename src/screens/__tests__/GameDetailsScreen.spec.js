import React from 'react';
import renderer from 'react-test-renderer';
import GameDetailsScreen from '../GameDetailsScreen';
import * as selectors from '../../utils/selectors';

jest.mock('../../components/Header', () => 'Header');
jest.mock('../../components/Paper', () => 'Paper');
jest.mock('../../components/DetailContent', () => 'DetailContent');

describe('GameDetailsScreen', () => {
  it('should renders correctly', () => {
    const props = {
      match: {
        params: {
          gameName: '94%'
        }
      }
    };
    const getGameDetailsByNameSpy = jest.fn().mockReturnValue({hero: 'hero', description:'description', name:'name', platform: 'platform'})
    selectors.getGameDetailsByName = getGameDetailsByNameSpy;
    const tree = renderer
      .create(<GameDetailsScreen {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})