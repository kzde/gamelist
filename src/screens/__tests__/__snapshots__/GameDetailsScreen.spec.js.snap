// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameDetailsScreen should renders NotFoundScreen if gameDetails is undefined 1`] = `<NotFoundScreen />`;

exports[`GameDetailsScreen should renders correctly 1`] = `
Array [
  <Header />,
  <div
    className="GameDetailsScreen__Media-hv417b-0 kkMeuk"
  >
    <img
      className="GameDetailsScreen__Img-hv417b-1 lgVoMk"
    />
    <div
      className="GameDetailsScreen__Shadow-hv417b-2 eGXpau"
    />
  </div>,
  <section
    className="GameDetailsScreen__DetailSection-hv417b-3 gbdMTe"
  >
    <Paper>
      <DetailContent
        content="description"
        subTitle="platform"
        title="name"
      />
    </Paper>
  </section>,
]
`;
